cmd/repo-updater/repos/github.go

```diff
@@ -382,6 +383,9 @@ func (s *GithubSource) listSearch(ctx context.Context, query string) (map[int64]
 	})
 }
 
+// regOrg is a regular expression that matches the pattern `org:<org-name>`
+var regOrg = regexp.MustCompile(`^org:(?P<org>[a-z0-9](?:-?[a-z0-9]){0,38})$`)
```

Comment:
Why the `38`?

Reply:
Is it the case that we only intend to support special casing of orgs in a `repositoryQuery` of this format: `org:$org`? If so, is a regular expression necessary or could we parse this with simpler mechanisms?

```go
ps := strings.SplitN(query, ":", 2)
if len(ps) == 2 && ps[0] == "org" {
    return s.listOrg(ctx, ps[1])
}
```

Reply:
Its a little more complicated than that. For example `org:foo bar` should not be special cased. I think sticking with a regex is the easiest way to do this.

However, your regex does not allow uppercase letters but org search does

Reply:
you can also put the `-` in the character class. So final suggested regex:

```suggestion
var regOrg = regexp.MustCompile(`^org:(?P<org>[a-zA-Z0-9][a-zA-Z0-9-]{0,38})$`)
```

Reply:
The regex expression matches GitHub's username convention.
- only hyphens and alphanumeric characters allowed
- can't begin/end with hyphen
- up to 38 characters
I'll add this in the comments.

Reply:
`^org:(?P<org>[a-zA-Z0-9][a-zA-Z0-9-]{0,38})$` allows for `org:sourcegraph-` so I just added `A-Z` to my original expression.